pandas/core/arrays/datetimelike.py

```diff
@@ -437,25 +690,122 @@ def _values_for_factorize(self):
     def _from_factorized(cls, values, original):
         return cls(values, dtype=original.dtype)
 
+    def _values_for_argsort(self):
+        return self._data
+
     # ------------------------------------------------------------------
-    # Null Handling
+    # Additional array methods
+    # These are not part of the EA API, but we implement them because
+    # pandas currently assumes they're there.
 
-    def isna(self):
-        return self._isnan
+    def view(self, dtype=None):
+        """
+        New view on this array with the same data.
 
-    @property  # NB: override with cache_readonly in immutable subclasses
-    def _isnan(self):
+        Parameters
+        ----------
+        dtype : numpy dtype, optional
+
+        Returns
+        -------
+        ndarray
+            With the specified `dtype`.
         """
-        return if each value is nan
+        return self._data.view(dtype=dtype)
+
+    def value_counts(self, dropna=False):
         """
-        return (self.asi8 == iNaT)
+        Return a Series containing counts of unique values.
 
-    @property  # NB: override with cache_readonly in immutable subclasses
-    def hasnans(self):
+        Parameters
+        ----------
+        dropna : boolean, default True
+            Don't include counts of NaT values.
+
+        Returns
+        -------
+        Series
         """
-        return if I have any nans; enables various perf speedups
+        # n.b. moved from PeriodArray.value_counts
+        from pandas import Series, Index
+
+        if dropna:
+            values = self[~self.isna()]._data
+        else:
+            values = self._data
+
+        cls = type(self)
+
+        result = value_counts(values, sort=False, dropna=dropna)
+        index = Index(cls(result.index, dtype=self.dtype),
+                      name=result.index.name)
+        return Series(result.values, index=index, name=result.name)
+
+    def searchsorted(self, value, side='left', sorter=None):
         """
-        return bool(self._isnan.any())
+        Find indices where elements should be inserted to maintain order.
```

Comment:
why is this not on EA base?

Reply:
Hasn't been proposed as far as I know. Personally, I would keep the scope small until people actually ask for things.

Reply:
Do we want to keep searchsorted on the the index class, to not bloat the API? Although Categorical has it...

Reply:
this for sure needs to be moved. I don't understand the motivation of NOT adding it. not to mention map, repeat. These are super common methods and just get re-implemented. Please do a pre-cursor PR for this.